### ISSUE TITLE
Crash report 7.4: NumberFormatException in EditPostActivity.onMediaUploadCancelClicked

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#5949]
```
Fatal Exception: java.lang.RuntimeException: Unable to destroy activity {org.wordpress.android/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.NumberFormatException: Invalid int: "null"
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5062)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5085)
       at android.app.ActivityThread.access$1700(ActivityThread.java:221)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1853)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7225)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.NumberFormatException: Invalid int: "null"
       at java.lang.Integer.invalidInt(Integer.java:138)
       at java.lang.Integer.parse(Integer.java:410)
       at java.lang.Integer.parseInt(Integer.java:367)
       at java.lang.Integer.parseInt(Integer.java:334)
       at java.lang.Integer.valueOf(Integer.java:525)
       at org.wordpress.android.ui.posts.EditPostActivity.onMediaUploadCancelClicked(EditPostActivity.java:2290)
       at org.wordpress.android.editor.EditorFragment.onDetach(EditorFragment.java:406)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1117)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1163)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1145)
       at android.app.FragmentManagerImpl.dispatchDestroy(FragmentManager.java:1998)
       at android.app.FragmentController.dispatchDestroy(FragmentController.java:218)
       at android.app.Activity.performDestroy(Activity.java:7096)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1170)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5040)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5085)
       at android.app.ActivityThread.access$1700(ActivityThread.java:221)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1853)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7225)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by maxme** [comment]
Note: this is new in 7.4

**Comment by tonyr59h** [comment]
#5673 could possibly be related.

**Comment by aforcier** [comment]
I'm seeing that ~70% of the reports for this issue are coming from API16, and for them this is a dup of https://github.com/wordpress-mobile/WordPress-Android/issues/5673, which is fixed in 7.5 (PR: https://github.com/wordpress-mobile/WordPress-Android/issues/5941).

For the rest, I'm seeing in the logs multiple instances of "trackLastActivity, activityId: Post Editor", and a few instances of Aztec being toggled off and on. I *think* part of the issue here is in migration of 'legacy' drafts.

I found a bug in the legacy draft migration which would end up setting the internal media upload id to `null` inside the visual editor - as far as I can see this is the only situation in which this can happen.

**Comment by mzorz** [comment]
Closed via #5996
